### PR TITLE
feat: DIA-1322: Use field_schema in skills as primary output format

### PR DESCRIPTION
**Pull request by niklub**
If Skill is provided with `field_schema` JSON, other output formatters like `output_template`, `labels` are ignored.

Additional changes in this PR:
- rewrite some tests to ensure they pass existing stubs (verifying nothing has changed)
- any `Runtime` execution now requires `response_model` (breaking change for those who use runtimes directly)

### CODE REVIEW

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/c18ad022d4ce12c18ad4c78710a124168e8f510c is created.

File: adala/skills/collection/entity_extraction.py [hunk 1]
```diff
@@ -6,37 +7,113 @@
...
+                    (k for k, v in schema["items"]["properties"].items()
+                     if v["type"] == "string" and "enum" in v and v['enum']), None)
+                if not self._labels_field_name:
+                    raise ValueError(f"EntityExtraction skill output field items properties must have one field"
+                                     f" of type string with enum (list of labels), got {schema['items']['properties']}")
+                # check if description provided for labels field, if not - warning and generate a default one
+                if "description" not in schema["items"]["properties"][self._labels_field_name]:
+                    logger.warning(f"EntityExtraction skill output field items properties labels field must have 'description', "
+                                   f"generated default description: {self._default_labels_description}")
+                    schema["items"]["properties"][self._labels_field_name]["description"] = self._default_labels_description
```
[the remark this answers is not in the document]
Comment (niklub) [reply adala/skills/collection/entity_extraction.py]: I put json schema validator for json schema, although it didn't make the code more comprehensive imo

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/2f057d1e5583d438d4b09849c7047ea8f58be34b is created.

**Review by hakan458 — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/be40eb98a9322e7701f633fa500a69a6ea991eb5 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/94721699b54ec6ec698e6280c2e75d618ccf724d is created.

**Pull request merged**